manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/ConnectorWorker.java

```diff
@@ -90,6 +95,17 @@ public ConnectorEntity createDependencies(Work work, ConnectorEntity connectorEn
             return connectorEntity;
         }
         if (status.getState() == ConnectorState.READY) {
+            // If the Connector is ready but differs to that required we need to patch it.
+            JsonNode existingConnectorDefinition = MAPPER.valueToTree(connector.getConnector());
+            JsonNode updatedConnectorDefinition = connectorEntity.getDefinition();
+            if (!Objects.equals(existingConnectorDefinition, updatedConnectorDefinition)) {
```

Comment:
Some connectors contain sensitive information which are not returned, i.e. Slack WebHook URL. As a result the `Objects.equals` will always return false, the Slack Processor will never change state to ready.

Reply:
Hmmm..that's a pain. Couple of initial thoughts here @manstis 

1. Have a `generation` field on the `ManagedResource` which is monotonically increased by 1 each time we accept an update from the user. This would let us know how many times a resource has been updated in total and would solve your problem here in that if the `generation` you're working on is > 1, then you know it has to be an update and a `patch` is required to RHOC. 
2. Have the ability to define an operation on the Work item e.g `CREATE`, `MODIFY`, `DELETE`
3. Something else :-)

I think I lean towards 1 as it's basically what the kube API does. Just some suggestions for you.

cheers,

Rob

Reply:
@sutaakar We don't seem to _hide_ any parameters at the moment; but the problem will become an issue with https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/755. I have tested this PR locally with a WebHook `Action` and successfully updated the URL.. but of course it will become a problem so I will look into a solution.

Reply:
@sutaakar @robpblake If you agree with my comment above; then it is best fixed in that PR where we can [retrieve](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/755/files#diff-fb73e1aeb94e2e7cdd2a9b560302e23913c137a3ca8d68c5bf4b3602044ff889R61-R71) *all* parameters (public and secret) and compare to that deployed to Managed Connectors. That seems the most viable solution. WDYT?

Reply:
@robpblake @sutaakar I think it may be best to handle the oncoming issue in the PR I referenced; where the issue is introduced. That PR [builds](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/8eeffd45d19018b5c7c589d877e08aa9c0e35f68/manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/ConnectorWorker.java#L133) the `Connector` definition from both the [public and secret parameters](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/8eeffd45d19018b5c7c589d877e08aa9c0e35f68/manager/src/main/java/com/redhat/service/smartevents/manager/connectors/ConnectorsServiceImpl.java#L75-L100) before invoking the RHOC API.

Reply:
@sutaakar 

> (that is the reason for failing E2E PR check)

Are you sure they fail for that reason? The change to hide _secret_ parameters isn't merged yet.

Reply:
@manstis The secret parameters are hidden by the RHOC API. It does not return sensitive parameters on the `GET` of the Connector details...

It would be more obvious if the value was returned masked or using some sort of redacted string e.g something like this:

```
{
  "sensitive_value": "******"
}
```

rather than

```
{
 "sensitive_value": ""
}
```

Reply:
> The secret parameters are hidden by the RHOC API. It does not return sensitive parameters on the GET of the Connector details...

Yikes. That's going to affect everything.